skdownscale/pointwise_models/gard.py

```diff
@@ -74,44 +79,83 @@ class AnalogRegression(AnalogBase):
         KDTree object
     """
 
-    def __init__(self, n_analogs=200, kdtree_kwargs=None, query_kwargs=None, lr_kwargs=None):
+    def __init__(
+        self,
+        n_analogs=200,
+        thresh=None,
+        kdtree_kwargs=None,
+        query_kwargs=None,
+        logistic_kwargs=None,
+        lr_kwargs=None,
+    ):
 
         self.n_analogs = n_analogs
+        self.thresh = thresh
         self.kdtree_kwargs = kdtree_kwargs
         self.query_kwargs = query_kwargs
+        self.logistic_kwargs = logistic_kwargs
         self.lr_kwargs = lr_kwargs
 
-    def predict(self, X):
+    def predict(self, X, return_errors=False, return_exceedance_prob=False):
         """ Predict using the AnalogRegression model
 
         Parameters
         ----------
         X : DataFrame, shape (n_samples, 1)
             Samples.
+        return_errors: boolean
+        return_exceedance_prob: boolean
 
         Returns
         -------
         C : pd.DataFrame, shape (n_samples, 1)
             Returns predicted values.
         """
+        # cannot return error and exceedance prob at the same time
+        assert not (return_errors is True and return_exceedance_prob is True)
+
         # validate input data
         check_is_fitted(self)
         X = check_array(X)
 
-        predicted = np.empty(len(X))
+        out = np.empty(len(X))
+
+        # not used if self.thresh = None, instantiating to keep the code clean
+        logistic_kwargs = default_none_kwargs(self.logistic_kwargs)
+        logistic_model = LogisticRegression(**logistic_kwargs)
```

Comment:
I think it would be okay to set `logistic_model = None` here since all of its uses fall under `if self.thresh` checks. But I could be missing something.